nemo_skills/evaluation/evaluator/math.py

```diff
@@ -58,30 +60,108 @@ class LeanEvaluatorConfig:
 
 
 def eval_lean4_proof(cfg):
-    eval_config = LeanEvaluatorConfig(**cfg.eval_config)
-
-    sandbox = get_sandbox(**eval_config.sandbox)
-    eval_config_dict = asdict(eval_config)
-    eval_config_dict.pop("sandbox")
-    asyncio.run(
-        sandbox.batch_evaluate_results(
-            input_files=cfg.input_files,
+    """Legacy function - delegates to Lean4ProofEvaluator class."""
+    evaluator = Lean4ProofEvaluator(cfg.eval_config)
+    asyncio.run(evaluator.eval_full(cfg.input_files))
+
+
+def eval_lean4_statement(cfg):
+    """Legacy function - delegates to Lean4StatementEvaluator class."""
+    evaluator = Lean4StatementEvaluator(cfg.eval_config)
+    asyncio.run(evaluator.eval_full(cfg.input_files))
+
+
+# Evaluator Classes
+
+
+class MathEvaluator(BaseEvaluator):
+    """Math evaluator - only supports batch evaluation."""
+
+    async def eval_full(self, input_files: list[str], **kwargs) -> None:
+        """Batch evaluate math problems."""
+        eval_config = MathEvaluatorConfig(**self.config)
+        eval_config_dict = asdict(eval_config)
+        batch_evaluate_results(
+            input_files=input_files,
+            **eval_config_dict,
+        )
+
+
+class Lean4ProofEvaluator(BaseEvaluator):
+    """Lean4 proof evaluator - supports both single and batch evaluation."""
+
+    async def eval_full(self, input_files: list[str], **kwargs) -> None:
+        """Batch evaluate Lean4 proofs."""
+        eval_config = LeanEvaluatorConfig(**self.config)
+        sandbox = get_sandbox(**eval_config.sandbox)
+        eval_config_dict = asdict(eval_config)
+        eval_config_dict.pop("sandbox")
+        await sandbox.batch_evaluate_results(
```

Comment:
ideally we don't use this but instead directly call eval_single in parallel. This will minimize code duplication